packages/taco/src/taco.ts

```diff
@@ -86,16 +139,47 @@ export const encrypt = async (
  *
  * @throws {Error} If the encryption process throws an error, an error is thrown.
  */
-export const encryptWithPublicKey = async (
+export async function encryptWithPublicKey(
   message: Uint8Array | string,
   condition: Condition,
   dkgPublicKey: DkgPublicKey,
   authSigner: ethers.Signer,
-): Promise<ThresholdMessageKit> => {
+): Promise<ThresholdMessageKit>;
+
+/**
+ * Encrypts a message with the given DKG public key under a specified condition.
+ *
+ * @export
+ * @param {Uint8Array | string} message  - The message to be encrypted.
+ * @param {Condition} condition - Condition under which the message will be encrypted. Those conditions must be
+ * satisfied in order to decrypt the message.
+ * @param {DkgPublicKey} dkgPublicKey - The public key of an active DKG Ritual to be used for encryption
+ * @param {Account} viemAuthSigner - The viem account that will be used to sign the encrypter authorization.
+ *
+ * @returns {Promise<ThresholdMessageKit>} Returns Promise that resolves with an instance of ThresholdMessageKit.
+ * It represents the encrypted message.
+ *
+ * @throws {Error} If the encryption process throws an error, an error is thrown.
+ */
+export async function encryptWithPublicKey(
+  message: Uint8Array | string,
+  condition: Condition,
+  dkgPublicKey: DkgPublicKey,
+  viemAuthSigner: Account,
+): Promise<ThresholdMessageKit>;
+
+export async function encryptWithPublicKey(
+  message: Uint8Array | string,
+  condition: Condition,
+  dkgPublicKey: DkgPublicKey,
+  authOrAccount: SignerLike,
```

Comment:
This leaks implementation:


```suggestion
  signer: SignerLike,
```